arangod/Scheduler/Scheduler.cpp

```diff
@@ -208,28 +212,76 @@ Scheduler::~Scheduler() {
 }
 
 // do not pass callback by reference, might get deleted before execution
-void Scheduler::post(std::function<void()> const& callback) {
+void Scheduler::post(std::function<void()> const callback, bool isV8,
+                     uint64_t timeout) {
   incQueued();
 
   try {
+    if (isV8) {
+      ++_queuedV8;
+    }
+
     // capture without self, ioContext will not live longer than scheduler
-    _ioContext.get()->post([this, callback]() {
-      JobGuard guard(this);
-      guard.work();
+    _ioContext->post([this, callback, isV8, timeout]() {
+      try {
+        if (isV8 && _queuedV8 > _maxQueuedV8 &&
+            numWorking(getCounters()) >= static_cast<uint64_t>(_maxQueuedV8)) {
+          --_queuedV8;
+          decQueued();
 
-      decQueued();
+          auto t = timeout;
 
-      callback();
+          if (t == 0) {
```

Comment:
This timeout magic is completely opaque...
Some comments would be helpful to understand why the timeout is bounded between the values `0` and `200`, and why a timeout value > 0 is doubled first.